internal/store/interfaces.go

```diff
@@ -8,37 +8,11 @@ import (
 
 //go:generate mockgen -source ./interfaces.go -destination ./mocks/interfaces.go -package mocks StatusTrackingGateway,Gateway,RouteTrackingListener,Listener,StatusTrackingRoute,Route,Store
 
-// StatusTrackingGateway is an optional extension
-// of Gateway. If supported by a Store, when
-// a Gateway is synced to an external location,
-// its corresponding callbacks should
-// be called.
-type StatusTrackingGateway interface {
-	Gateway
-
-	TrackSync(ctx context.Context, sync func() (bool, error)) error
```

Comment:
This and `SyncStatus` get wrapped into a `StatusUpdater` interface that's optional to set for the concrete store implementation -- that way we can drop the overloaded `StatusTrackingGateway` and `StatusTrackingRoute`